csrc/codegen.cpp

```diff
@@ -2680,7 +2709,7 @@ class CudaKernelGenerator : private OptOutConstDispatch {
     }
   }
 
-  void handle(const kir::BlockSync* sync) final {
+  void handle(kir::BlockSync* sync) final {
     // Use a custom synchronization method if enabled
     if (std::getenv("PYTORCH_NVFUSER_USE_BLOCK_SYNC_ATOMIC")) {
       indent() << "block_sync::sync();\n";
```

Comment:
Does it make sense to do a further cleanup of `BlockSync` to remove `isAligned` from it, and remove `ConvertAlignedBlockSync` completely, and here just generate `f'block_sync::sync<{isAligned()}>()'`?

Reply:
I think so, the only reason why we do the conversion was to get better matmul perf, I think this goal can still be achieved with the suggest change.

Reply:
Done